File: Grind75/maximumSubarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, maxSum = INT_MIN;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            sum = max(nums[i], sum);
            maxSum = max(sum, maxSum);
        }
        return maxSum;
    }
};

File: maximumSubarray.py
from typing import List

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        summ,maxSum = 0,-2**31
        for num in nums:
            summ += num
            if summ < num:
                summ = num
            if maxSum < summ:
                maxSum = summ
                
        return maxSum